internal/providers/terraform/aws/efs_file_system.go

```diff
@@ -10,34 +12,60 @@ func GetEFSFileSystemRegistryItem() *schema.RegistryItem {
 	return &schema.RegistryItem{
 		Name:  "aws_efs_file_system",
 		RFunc: NewEFSFileSystem,
+		Notes: []string{"In Terraform config the availability_zone_name field is required in this resource for correct results"},
 	}
 }
 
 func NewEFSFileSystem(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
 	region := d.Get("region").String()
-
+	var availabilityZoneName string
 	costComponents := make([]*schema.CostComponent, 0)
+	var gbStorage, infrequentAccessGbStorage *decimal.Decimal
 
-	var gbStorage *decimal.Decimal
-	if u != nil && u.Get("storage_gb").Exists() {
+	if u != nil && u.Get("infrequent_access_storage_gb").Type != gjson.Null {
+		infrequentAccessGbStorage = decimalPtr(decimal.NewFromFloat(u.Get("infrequent_access_storage_gb").Float()))
+	}
+	if u != nil && u.Get("storage_gb").Type != gjson.Null {
 		gbStorage = decimalPtr(decimal.NewFromFloat(u.Get("storage_gb").Float()))
 	}
 
-	costComponents = append(costComponents, &schema.CostComponent{
-		Name:            "Storage (standard)",
-		Unit:            "GB",
-		UnitMultiplier:  1,
-		MonthlyQuantity: gbStorage,
-		ProductFilter: &schema.ProductFilter{
-			VendorName:    strPtr("aws"),
-			Region:        strPtr(region),
-			Service:       strPtr("AmazonEFS"),
-			ProductFamily: strPtr("Storage"),
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "usagetype", ValueRegex: strPtr("/-TimedStorage-ByteHrs/")},
-			},
-		},
-	})
+	if d.Get("availability_zone_name").Type != gjson.Null {
+		availabilityZoneName = d.Get("availability_zone_name").String()
+		if strings.HasPrefix(strings.ToLower(availabilityZoneName), "standard") {
+			costComponents = append(costComponents, &schema.CostComponent{
```

Comment:
I think it would be nice if you move this cost component into a separate function and set the name, usage type and quantity as parameters.

Reply:
Done